components/higher-order/with-fallback-styles/index.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * External dependencies
+ */
+import { findKey, isEqual } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { Component } from '@wordpress/element';
+
+export default ( mapNodeToProps ) => ( WrappedComponent ) => {
+	return class extends Component {
+		constructor() {
+			super( ...arguments );
+			this.nodeRef = null;
+			this.state = {
+				fallbackStyles: undefined,
+				grabStylesCompleted: false,
+			};
+
+			this.bindRef = this.bindRef.bind( this );
+		}
+
+		bindRef( node ) {
+			if ( ! node ) {
+				return;
+			}
+			this.nodeRef = node;
+		}
+
+		componentDidMount() {
+			this.grabFallbackStyles();
+		}
+
+		componentDidUpdate() {
+			this.grabFallbackStyles();
+		}
+
+		grabFallbackStyles() {
+			const { grabStylesCompleted, fallbackStyles } = this.state;
+			if ( this.nodeRef && ! grabStylesCompleted ) {
```

Comment:
Can you explain why we need to use `grabStylesCompleted` in here? Can we check if `fallbackStyles` is still `undefined` instead? It looks like you can always compute `grabStylesCompleted` using the existing variables.

Reply:
Hi @gziolo, for example in button we may have set a custom text color, so we can not compute the fallbackTextColor but we can compute the fallbackBackgroudColor. So our mapNodeToProps returns undefined for the properties it cannot compute the fallback color (fallbackTextColor). When there is no undefined value ( ! findKey( newFallbackStyles, ( obj ) => obj === undefined ) ) = true we know we don't need to grab colors anymore. The value of this is saved in grabStylesCompleted so we are not running this line with findKey all the time.

Another advantage is when inserting a component no custom colors are set so all fallback colors are immediately saved and logic to grab color is just executed one time.

Reply:
Would it work if you would use [every](https://lodash.com/docs/#every) from lodash instead of findKey:
‘’’js
every( newFallbackStyles )
‘’’
It might better express intent.

Reply:
I agree using "every" would make things simpler to understand, but I think we can not use it here. Every works on a collection here we are working a single object and we just want to make sure there is no key/property of that object that is undefined.

Reply:
This is a perfectly fine usage
```js
every( { a: true, b: 1, c: undefined } );
````
It works with with both arrays and object:
> collection (Array|Object): The collection to iterate over.

Some utilty methods that operate on collections, work with strings, too. See [includes](https://lodash.com/docs/4.17.4#includes).

Reply:
Thank you for your suggestion @gziolo you were right and your suggestion was applied before merging.